### ISSUE TITLE
IntervalArithmetic.Interval(::Float64, ::Float64) appears no longer supported in v0.21.0; replace with IntervalArithmetic.interval()

### ISSUE THREAD

**Issue by vtjeng** [JuliaIntervals/IntervalArithmetic.jl#581]
**tl;dr**: Replace `Interval` with `interval`.

---

I maintain a package that uses `IntervalArithmetic` and tests are currently failing when I attempt to use `v0.21.0`. The lines of code causing the failure in my package can be found here: 

https://github.com/vtjeng/MIPVerify.jl/blob/678ef7095bd42cff940b9cc25980b45b62809cee/src/vendor/ConditionalJuMP.jl#L26-L34

It looks like the commit that can be found here, where `Interval()` is removed: https://github.com/JuliaIntervals/IntervalArithmetic.jl/commit/98687e44a68087848fc66d2919e07e37cc41f63a#diff-1fdf1d32d926756fb9352a58498e775094262f78140cd4fe3a739477d213623fL79-L82

You can reproduce the issue on the command line.

```
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.7.1 (2021-12-22)
 _/ |\__'_|_|_|\__'_|  |  Official https://julialang.org/ release
|__/                   |

julia> using IntervalArithmetic
 │ Package IntervalArithmetic not found, but a package named IntervalArithmetic is available from a registry.
 │ Install package?
 │   (@v1.7) pkg> add IntervalArithmetic
 └ (y/n) [y]: y

[...]

julia> IntervalArithmetic.Interval(5, 5)
[5, 5]
```

```
$ ~/Downloads/julia-1.8.5/bin/julia
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.8.5 (2023-01-08)
 _/ |\__'_|_|_|\__'_|  |  Official https://julialang.org/ release
|__/                   |

julia> using IntervalArithmetic
 │ Package IntervalArithmetic not found, but a package named IntervalArithmetic is available from a registry.
 │ Install package?
 │   (@v1.8) pkg> add IntervalArithmetic
 └ (y/n/o) [y]: y
[...]
julia> IntervalArithmetic.Interval(5.0, 5.0)
ERROR: MethodError: no method matching Interval(::Float64, ::Float64)
Closest candidates are:
  (::Type{T})(::T) where T<:Number at boot.jl:772
Stacktrace:
 [1] top-level scope
   @ REPL[3]:1
```

**Comment by OlivierHnt** [comment]
In v0.21 of IntervalArithmetic, the supported Julia version is v1.8 and above; this explains why a fresh install of IntervalArithmetic is different in v1.7 and in v1.8 of Julia.

About the removal of `Interval`: 0.21 and master contains breaking changes in an attempt to minimise silent errors. Conversion/promotion system of Julia is typically triggered when using generic code and various input types are being combined; this is the type of dangerous zone we want to avoid. So since for a struct `A <: Number` conversion is equivalent to `A(x)`, we removed all default constructors.
Now, the go-to constructor is `interval`; e.g. `interval(Float64, 1, 2)`, or `interval(1, 2)`, or `interval(1)`.

From my understanding this issue falls into the scope of #580; so I am closing it as a duplicate, but please re-open it if I misunderstood the situation 🙂.

**Comment by vtjeng** [comment]
@OlivierHnt - yes, that lines up with what I'm thinking. The point of this issue was to flag it for other developers who relied on the `Interval` constructor so they didn't have to go through the search process I did. 😃
